bin/logstash.lib.sh

```diff
@@ -1,5 +1,29 @@
 unset CDPATH
-LOGSTASH_HOME=$(cd `dirname $0`/..; pwd)
+# This unwieldy bit of scripting is to try to catch instances where Logstash
+# was launched from a symlink, rather than a full path to the Logstash binary
+if [ -L $0 ]; then
```

Comment:
I considered that, but the error messages are slightly different.  Also, inclusion works with bash, but not as simply with bourne shell (which this uses for its universality).